### ISSUE TITLE
Managed Identity token expired 

### ISSUE THREAD

**Issue by alexpoontm** [microsoft/azure-container-apps#543]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is a: (mark with an x)
- [x] bug report -> please search issues before submitting
- [x] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)

### Issue description
Hi, I am new in container app. My applications are using Dapr which required to access blob storage as state store. I followed [this offical document](https://learn.microsoft.com/en-us/azure/container-apps/microservices-dapr?tabs=bash#set-up-a-state-store) and assigned a managed identity(user assigned) with storage acount contributor role to the container app. Everything works great within 24hours but failed after that.

### Steps to reproduce
1. follow  [this offical document](https://learn.microsoft.com/en-us/azure/container-apps/microservices-dapr?tabs=bash#set-up-a-state-store) to deploy app container
2. wait for 24hours 

**Expected behavior**
The app container should take care the token refreshing? or the document should mention about how to refesh token if it expired?

**Actual behavior** 
Result below error after 24hours
**AuthenticationErrorDetail: Lifetime validation failed. The token is expired.**
```
2022-12-23T03:51:03.184257498Z       System.AggregateException: One or more errors occurred. (One or more errors occurred. (State operation failed: the Dapr endpoint indicated a failure. See InnerException for details.))
2022-12-23T03:51:03.184264998Z        ---> System.AggregateException: One or more errors occurred. (State operation failed: the Dapr endpoint indicated a failure. See InnerException for details.)
2022-12-23T03:51:03.184270099Z        ---> Dapr.DaprException: State operation failed: the Dapr endpoint indicated a failure. See InnerException for details.
2022-12-23T03:51:03.184279199Z        ---> Grpc.Core.RpcException: Status(StatusCode="Internal", Detail="fail to get Tenant from state store commonstatestore: -> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, /home/vsts/work/1/go/pkg/mod/github.com/!azure/azure-storage-blob-go@v0.10.0/azblob/zc_storage_error.go:42
2022-12-23T03:51:03.184303099Z       ===== RESPONSE ERROR (ServiceCode=AuthenticationFailed) =====
2022-12-23T03:51:03.184308799Z       Description=Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.
2022-12-23T03:51:03.184314199Z       RequestId:9f4cab3a-201e-0140-2381-160eb6000000
2022-12-23T03:51:03.184319599Z       Time:2022-12-23T03:51:03.1812983Z, Details: 
2022-12-23T03:51:03.184324699Z          AuthenticationErrorDetail: Lifetime validation failed. The token is expired.
2022-12-23T03:51:03.184329999Z          Code: AuthenticationFailed
2022-12-23T03:51:03.184335199Z          GET https://sqlva5awuuhacgad74.blob.core.windows.net/state/Tenant?timeout=61
2022-12-23T03:51:03.184340499Z          Authorization: REDACTED
2022-12-23T03:51:03.184345899Z          User-Agent: [dapr-1.9.5-msft-1 Azure-Storage/0.10 (go1.19; linux)]
2022-12-23T03:51:03.184350899Z          X-Ms-Client-Request-Id: [6b64dc9c-ffe8-4b67-6de2-46195099ba3e]
2022-12-23T03:51:03.184356099Z          X-Ms-Version: [2019-02-02]
2022-12-23T03:51:03.184361099Z          --------------------------------------------------------------------------------
2022-12-23T03:51:03.184366499Z          RESPONSE Status: 403 Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.
2022-12-23T03:51:03.184374499Z          Content-Length: [424]
2022-12-23T03:51:03.184379399Z          Content-Type: [application/xml]
2022-12-23T03:51:03.184384699Z          Date: [Fri, 23 Dec 2022 03:51:02 GMT]
2022-12-23T03:51:03.184390099Z          Server: [Microsoft-HTTPAPI/2.0]
2022-12-23T03:51:03.184394299Z          X-Ms-Error-Code: [AuthenticationFailed]
2022-12-23T03:51:03.184398199Z          X-Ms-Request-Id: [9f4cab3a-201e-0140-2381-160eb6000000]
2022-12-23T03:51:03.184402399Z       
2022-12-23T03:51:03.184406199Z       
2022-12-23T03:51:03.184409699Z       ")
2022-12-23T03:51:03.184413300Z          at Dapr.Client.DaprClientGrpc.GetStateAsync[TValue](String storeName, String key, Nullable`1 consistencyMode, IReadOnlyDictionary`2 metadata, CancellationToken cancellationToken)
```

**Comment by anthonychu** [comment]
@ItalyPaleAle @vturecek Do you know what might be happening here?

**Comment by TonyFjo** [comment]
We see the exact same behaviour but to storage, any update on this one?

type: 'DaprException',
2023-01-23T08:59:34.968026642Z     title: 'State operation failed: the Dapr endpoint indicated a failure. See InnerException for details.',
2023-01-23T08:59:34.968030542Z     status: 500,
2023-01-23T08:59:34.968035542Z     detail: 'Status(StatusCode="Internal", Detail="fail to get 236efa75-e400-4883-a6f7-c56e31c24149 from state store tenant-zones: -> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, /home/vsts/work/1/go/pkg/mod/github.com/!azure/azure-storage-blob-go@v0.10.0/azblob/zc_storage_error.go:42\n' +
2023-01-23T08:59:34.968039442Z       '===== RESPONSE ERROR (ServiceCode=AuthenticationFailed) =====\n' +
2023-01-23T08:59:34.968043942Z       'Description=Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.\n' +
2023-01-23T08:59:34.968055342Z       'RequestId:016addc2-f01e-003c-7d09-2f18ce000000\n' +
2023-01-23T08:59:34.968066042Z       'Time:2023-01-23T08:59:34.9513095Z, Details: \n' +
2023-01-23T08:59:34.968069942Z       '   AuthenticationErrorDetail: Lifetime validation failed. The token is expired.\n' +
2023-01-23T08:59:34.968077642Z       '   Code: AuthenticationFailed\n' +
2023-01-23T08:59:34.968086742Z       '   GET https://<our storagename>.blob.core.windows.net/tenant-zones/236efa75-e400-4883-a6f7-c56e31c24149?timeout=61\n' +
2023-01-23T08:59:34.968094342Z       '   Authorization: REDACTED\n' +
2023-01-23T08:59:34.968098242Z       '   User-Agent: [dapr-1.9.5-msft-1 Azure-Storage/0.10 (go1.19; linux)]\n' +
2023-01-23T08:59:34.968105642Z       '   X-Ms-Client-Request-Id: [c2453879-0d93-4487-6070-f581d56e9dc9]\n' +
2023-01-23T08:59:34.968113342Z       '   X-Ms-Version: [2019-02-02]\n' +
2023-01-23T08:59:34.968117342Z       '   --------------------------------------------------------------------------------\n' +
2023-01-23T08:59:34.968128742Z       '   RESPONSE Status: 403 Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.\n' +
2023-01-23T08:59:34.968136442Z       '   Content-Length: [424]\n' +
2023-01-23T08:59:34.968144142Z       '   Content-Type: [application/xml]\n' +
2023-01-23T08:59:34.968152042Z       '   Date: [Mon, 23 Jan 2023 08:59:34 GMT]\n' +
2023-01-23T08:59:34.968155742Z       '   Server: [Microsoft-HTTPAPI/2.0]\n' +
2023-01-23T08:59:34.968162841Z       '   X-Ms-Error-Code: [AuthenticationFailed]\n' +
2023-01-23T08:59:34.968170441Z       '   X-Ms-Request-Id: [016addc2-f01e-003c-7d09-2f18ce000000]\n' +
2023-01-23T08:59:34.968174541Z       '\n' +
2023-01-23T08:59:34.968182241Z       '\n' +
2023-01-23T08:59:34.968186741Z       '")',

**Comment by ItalyPaleAle** [comment]
The issue was fixed with Dapr 1.9.5-msft-2 which should be available in all Azure regions
